js/maps/tiles.js

```diff
@@ -1,32 +1,125 @@
-// This file only contains lists of spells and their actions / attributes
+// This file only contains lists of tiles and their actions / attributes
 if(window.app == null ||
    window.app == undefined) {
   window.app = {};
 }
 
 var TILES = {
+// Basic Environment Tiles
   DIRT: {
     name: "DIRT",
     imageUrl: "", 
     solid: false,
     action: null
-    info: 
+    info: "Just some dirt",
   },
 
-  SHOP_DOOR: {
-    name: "DIRT",
+  GRASS: {
+    name: "GRASS",
     imageUrl: "", 
     solid: false,
-    action: function(){
+    action: null
+    info: "Just some grass",
+  },
+
+  RIVER: {
+    name: "RIVER",
+    imageUrl: "", 
+    solid: true,
+    action: null
+    info: "Just some water",
+  },
+
+  GARDEN: {
+    name: "GARDEN",
+    imageUrl: "",
+    solid: false,
+    action: null,
+    info: "A small garden"
+  }
 
+  WELL: {
+    name: "WELL",
+    imageUrl: "",
+    solid: true,
+    action: null,
+    info: "A small well"
+  }
+
+// Fixture Tiles
+  SHOP_DOOR: {
+    name: "SHOP_DOOR",
+    imageUrl: "", 
+    solid: true,
+    action: function(){
+      // Open Shop?
     }
+    info: "",
   },
 
-  DOOR: {
-    name: "DOOR".
+  CLOSED_DOOR: {
```

Comment:
Maybe we should make doors a bit more cleaver than the rest of the tiles. The `action` can invoke other methods that change the image URL and it's state. What we've done (essentially) by requiring `name`, `imageUrl`, `solid`, `action`, and `info` was make an interface.  A "promise" that these attributes / methods will exist.  But that doesn't limit an object to only those attributes and (private) methods.